lib/datadog/tracing/contrib/patcher.rb

```diff
@@ -48,6 +53,12 @@ def on_patch_error(e)
             # Log the error
             Datadog.logger.error("Failed to apply #{patch_name} patch. Cause: #{e} Location: #{Array(e.backtrace).first}")
 
+            @patch_error_result = {
+              type: e.respond_to?(:class) ? e.class.name : nil,
+              message: e.respond_to?(:message) ? e.message : nil,
+              line: e.respond_to?(:backtrace) ? Array(e.backtrace).first : nil
```

Comment:
No need to check `#respond_to?` here. `e` is an [`Exception`](https://ruby-doc.org/core-2.5.1/Exception.html), and all these methods will always be present.